commons/zenoh-buffers/src/vec.rs

```diff
@@ -103,7 +103,7 @@ impl Writer for &mut Vec<u8> {
         let s = crate::unsafe_slice_mut!(self.spare_capacity_mut(), ..len);
         // SAFETY: converting MaybeUninit<u8> into [u8] is safe because we are going to write on it.
```

Comment:
This is backwards. We should write and to it and _then_ `MaybeUninit::assume_init`.

Reply:
Tackling the uninitialized memory issue not the subject of this PR.